Sources/StreamChat/Query/LocalFilterProcessor.swift

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+struct LocalFilterProcessor<
+    Scope: FilterScope,
+    ResultType
+> {
+    var filter: Filter<Scope>
+
+    func execute(
+        _ items: [ResultType]
+    ) -> [ResultType] {
+        guard
+            filter.key != nil,
+            let `operator` = FilterOperator(rawValue: filter.operator),
+            let keyToValueMapper = filter.keyToValueMapper
+        else {
+            return items
+        }
+
+        let mappedValues: [(item: ResultType, value: AnyFilterValue)] = items
+            .compactMap {
+                guard let value = keyToValueMapper($0) else { return nil }
+                return (item: $0, value: value.eraseToAnyFilterValue())
+            }
+        let anyValue = filter.value.eraseToAnyFilterValue()
+
+        switch `operator` {
+        case .equal:
+            return mappedValues.filter { $0.value == anyValue }.map(\.item)
+        case .notEqual:
+            return mappedValues.filter { $0.value != anyValue }.map(\.item)
+        case .greater:
+            return mappedValues.filter { $0.value > anyValue }.map(\.item)
+        case .greaterOrEqual:
+            return mappedValues.filter { $0.value >= anyValue }.map(\.item)
+        case .less:
+            return mappedValues.filter { $0.value < anyValue }.map(\.item)
+        case .lessOrEqual:
+            return mappedValues.filter { $0.value <= anyValue }.map(\.item)
+        case .in:
+            var result: [ResultType] = []
+            if let inputValueSet = anyValue.valueSet {
+                for mappedValue in mappedValues {
+                    guard let mappedValueSet = mappedValue.value.valueSet else {
+                        continue
+                    }
+                    if inputValueSet.intersection(mappedValueSet) == inputValueSet {
+                        result.append(mappedValue.item)
+                    }
+                }
+            }
+            return result
+        case .notIn:
+            var result: [ResultType] = []
+            if let inputValueSet = anyValue.valueSet {
+                for mappedValue in mappedValues {
+                    guard let mappedValueSet = mappedValue.value.valueSet else {
+                        continue
+                    }
+                    if inputValueSet.subtracting(mappedValueSet) == inputValueSet {
+                        result.append(mappedValue.item)
+                    }
+                }
+            }
+            return result
+        case .query:
+            //            return "\(key) QUERY \(value)"
+            return []
```

Comment:
We can ignore this one. This is just used for searches 👍 WS events are not important there